### PR TITLE
Retrofit secrets detection to use scripts/detect_secrets_baseline.sh

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Retrofits secrets detection to use the shared `scripts/detect_secrets_baseline.sh` pattern from `template-repo-java`, replacing the previous inline GitHub Actions script with a consistent local + CI workflow.

**Changes:**
- Add `scripts/detect_secrets_baseline.sh` — single source of truth for `detect-secrets` args with `scan`, `audit`, and default (CI check) modes; JS-specific excludes (`node_modules`, `dist`, `build`)
- Add `scripts/requirements.txt` — pins `detect-secrets~=1.5.0`
- Add `.detect-secrets-ignore` — per-repo exclusion patterns (`README.md`, `CHANGELOG.md`)
- Update `.secrets.baseline` to v1.5.0 format with all existing email-address findings marked as audited (`is_secret: false`)
- Update `secrets-detection.yaml` — now installs via `scripts/requirements.txt` and delegates all scan logic to the script; adds `if: github.actor != 'pdsen-ci'` guard and baseline-existence pre-check
- Update `README.md` — adds **Secret Detection** section with prerequisites, usage commands, and developer workflow

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90%

## ⚙️ Test Data and/or Report

The updated `secrets-detection.yaml` workflow will run on this PR against `main`. The `.secrets.baseline` has been updated to v1.5.0 format with all prior findings audited, so the unaudited-count check should pass.

Local verification:
```bash
pip install -r scripts/requirements.txt
scripts/detect_secrets_baseline.sh   # should exit 0
```

## ♻️ Related Issues

N/A

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: scripts/detect_secrets_baseline.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    echo "Updated .secrets.baseline"
+    echo "Next step: run 'scripts/detect_secrets_baseline.sh audit' to review and classify detected secrets."
+elif [ "$1" = "audit" ]; then
+    $DETECT_SECRETS audit .secrets.baseline
+else
+    # Check 1: Fail if any secrets in the baseline have not been audited
+    unaudited=$(python3 -c "
+import json, sys
+with open('.secrets.baseline') as f: data = json.load(f)
+count = sum(1 for v in data.get('results', {}).values() for s in v if 'is_secret' not in s)
```
Comment (nutjob4life) [inline scripts/detect_secrets_baseline.sh]: Ahh, there's subtle edge case here. `is_secret` is effectively a three-state value.  In the baseline file you typically see:
```json
"is_secret": true (or false)
```
But before a finding has been audited, the field is absent entirely.

Checking only for the _presence_ of `is_secret` collapses those three states to two:

- Missing `is_secret`: _fail_ because it has not been audited
- `is_secret: true`: _fail_ because it's a confirmed real credential
- `is_secret: false`: _permit_ because it's an audited false positive

. First the **unaudited** findings:
```python
unaudited = sum(
    1
    for v in data.get("results", {}).values()
    for s in v
    if "is_secret" not in s
)
```
Then the **confirmed secrets**:
```python
confirmed = sum(
    1
    for v in data.get("results", {}).values()
    for s in v
    if s.get("is_secret") is True
)
```
Then the script becomes:
```bash
if [[ "$unaudited" -gt 0 ]]; then
    echo "$unaudited finding(s) have not been audited."
    exit 1
fi

if [[ "$confirmed" -gt 0 ]]; then
    echo "$confirmed real secret(s) are present in the baseline."
    exit 1
fi
```

**Review by nutjob4life — APPROVED** [review]
This is, as before, _fabulous_ to see. Thanks for making secrets detection 100× better.

One possible small minor edge case; see the comment on `is_secret` in the `detect_secrets_baseline.sh` script. Deal-breaker? I don't know. What do you think?

Oh an SonarQube had a few `[…]` → `[[…]]` things too.

But despite all that I'm giving my stamp: ✅

File: scripts/detect_secrets_baseline.sh [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+DETECT_SECRETS="$(find_detect_secrets)"
+
+# ---------------------------------------------------------------------------
+# Build --exclude-files arguments from global + per-repo patterns
+# ---------------------------------------------------------------------------
+
+# Paths that are always excluded, regardless of repo type.
+GLOBAL_EXCLUDES=(
+    '\.secrets\..*'             # the baseline files themselves
+    '\.git.*'                   # git internals
```
Comment (nutjob4life) [inline scripts/detect_secrets_baseline.sh]: This regex could also block detection to things like

    .github/workflows/secret-detection.yaml
    .github/dependabot.yml

and other things in `.github`, which may be a place where secrets could crop up.

Consider path-segment anchoring?

    '(^|/)\.git/'

**Review by nutjob4life — APPROVED** [review]
10/10 would merge and merge again! No notes!

Well, one note; see the comment about "regex" on 

    '\.git.*'  # git internals

but otherwise good to go! ✅
